### ISSUE TITLE
Bikeshed seems to incorrectly use the last occurence of type for method argument validation

### ISSUE THREAD

**Issue by bialpio** [tabatkins/bikeshed#1566]
When attempting to generate a spec from the below .bs file, bikeshed raises an error `FATAL ERROR: IDL ERROR LINE: 5 - Dictionary argument "init" without required members must be marked optional`. The error goes away either when the entire `partial dictionary FakeXRDeviceInit` gets uncommented, when making `FakeXRDeviceInit init` argument to `simulateDeviceConnection` method optional, or when adding some required key to the `partial dictionary FakeXRDeviceInit`. It seems that when validating method arguments, bikeshed only takes into account the latest occurrence of the `FakeXRDeviceInit` dictionary & checks if it has any required keys, instead of looking at all the keys present in the definition (including the ones introduced by partial definitions).  

```
<pre class="metadata">
Status: DREAM
Shortname: bikeshed-sandbox
Title: Bikeshed Sandbox
Editor: you
 
Abstract: Used for experiments with bikeshed.
</pre>
 
<script type="idl">
interface FakeXRDevice {};
 
interface XRTest {
  Promise<FakeXRDevice> simulateDeviceConnection(FakeXRDeviceInit init);
};
 
dictionary FakeXRDeviceInit {
    required boolean supportsImmersive;
};
 
partial dictionary FakeXRDeviceInit {
};
 
</script>
```

**Comment by tabatkins** [comment]
Indeed, pretty certain that's exactly what I do; within a single spec you shouldn't be defining something both as a base and as a partial. ^_^

That said, I suspect the same problem would occur if a spec extended a dictionary defined elsewhere, and then used it.

**Comment by bialpio** [comment]
Thanks for confirming! I've unblocked myself by moving the keys that I wanted to introduce in a partial definition up to the main definition. This might be more difficult to work around in multi-spec scenario though.

**Comment by cabanier** [comment]
I'm also running in this issue. 
In a new spec, I'm defining a dictionary that has required members and then I have 3 dictionaries that inherit from it and that have no required members.
Then, when I compile the spec I get:

> FATAL ERROR: IDL ERROR LINE: 119 - Dictionary argument "init" without required members must be marked optional

>
